server/channels/store/sqlstore/property_group_store.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	sq "github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+)
+
+var propertyGroupColumns = []string{"ID", "Name"}
+
+type SqlPropertyGroupStore struct {
+	*SqlStore
+}
+
+func newPropertyGroupStore(sqlStore *SqlStore) store.PropertyGroupStore {
+	return &SqlPropertyGroupStore{sqlStore}
+}
+
+func (s *SqlPropertyGroupStore) Register(name string) (*model.PropertyGroup, error) {
+	if name == "" {
+		return nil, store.NewErrInvalidInput("PropertyGroup", "name", name)
+	}
+
+	existingGroup, err := s.Get(name)
+	if err == nil {
+		return existingGroup, nil
+	}
+
+	var nfErr *store.ErrNotFound
+	if !errors.As(err, &nfErr) {
+		return nil, errors.Wrap(err, "property_group_register")
+	}
```

Comment:
This is not atomic. Imagine 2 concurrent Register requests. One will pass, one will fail. We need to use an `INSERT .. ON CONFLICT` clause for this. See `(ls SqlLicenseStore) Save`.

Reply:
Fixed, PTAL f48d8a126d